openstack/identity/v3/extensions/endpointgroups/results.go

```diff
@@ -0,0 +1,76 @@
+package endpointgroups
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+type commonResult struct {
+	gophercloud.Result
+}
+
+// Extract interprets a GetResult, CreateResult or UpdateResult as a concrete
+// EndpointGroup. An error is returned if the original call or the extraction
+// failed.
+func (r commonResult) Extract() (*EndpointGroup, error) {
+	var s struct {
+		EndpointGroup *EndpointGroup `json:"endpoint_group"`
+	}
+	err := r.ExtractInto(&s)
+	return s.EndpointGroup, err
+}
+
+// GetResult is the response from a Get operation. Call its Extract method
+// to interpret it as an EndpointGroup.
+type GetResult struct {
+	commonResult
+}
+
+// EndpointFilter represents a set of one or several criteria to match endpoints.
+type EndpointFilter struct {
+	// Availability is the interface type to filter (admin, internal,
+	// or public), referenced by the gophercloud.Availability type.
+	Availability gophercloud.Availability `json:"interface,omitempty"`
+
+	// ServiceID is the ID of the service to filter
+	ServiceID string `json:"service_id,omitempty"`
+
+	// RegionID is the ID of the region to filter
+	RegionID string `json:"region_id,omitempty"`
+}
+
+// EndpointGroup represents a group of endpoints matching one or several filter
+// criteria.
+type EndpointGroup struct {
+	// ID is the unique ID of the endpoint group.
+	ID string `json:"id"`
+
+	// Name is the name of the new endpoint group.
+	Name string `json:"name"`
+
+	// Filters is an EndpointFilter type describing the filter criteria
+	Filters EndpointFilter `json:"filters"`
```

Comment:
I'm thinking we want to do a `map[string]interface{}` here instead of a well-defined `EndpointFilter`. 

The filter keys you linked to (https://github.com/openstack/keystone/blob/ba2e4b83e88d5c1576e9d2cd6a760367deef3a62/keystone/api/os_ep_filter.py#L54) appear to be for POST/creation validation and not defining what the actual data structure is.

The schema looks to be defining this field as a generic JSON blob (https://github.com/openstack/keystone/blob/ba2e4b83e88d5c1576e9d2cd6a760367deef3a62/keystone/catalog/backends/sql.py#L614).

It's very possible that right now only the `valid_filter_keys` are the only 3 entries, but since the schema is defining this as a generic JSON blob, I think we should just use `map[string]interface{}` to reflect that.

Thoughts?